chunky/src/java/se/llbit/chunky/resources/pbr/LabPbrSpecularMap.java

```diff
@@ -0,0 +1,143 @@
+package se.llbit.chunky.resources.pbr;
+
+import se.llbit.chunky.resources.BitmapImage;
+import se.llbit.math.Ray;
+
+/**
+ * Specular map that follows the labPBR format.
+ *
+ * @see <a href="https://github.com/rre36/lab-pbr/wiki/Specular-Texture-Details#alpha">Specular
+ * Texture details</a>
+ */
+public class LabPbrSpecularMap implements EmissionMap, ReflectanceMap, RoughnessMap, MetalnessMap {
+
+  private final int width;
+  private final int height;
+  private byte[] emissionMap;
+  private byte[] reflectanceMap;
+  private float[] roughnessMap;
+  private boolean hasMetalness;
+
+  public LabPbrSpecularMap(BitmapImage texture) {
+    width = texture.width;
+    height = texture.height;
+
+    emissionMap = new byte[texture.width * texture.height];
+    boolean hasEmission = false;
+    for (int y = 0; y < texture.height; ++y) {
```

Comment:
We loop over the texture three times here. Can this be refactored so that we only loop once?